dt command: no type code for std::nullptr_t

### Description
The `dt` command throws an exception on `dt std::nullptr_t`.

### Steps to reproduce
Compile any cpp program.

```
pwndbg> dt std::nullptr_t
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ ------------------------/pwndbg/pwndbg/commands/__init__.py:189 in __call__                      │
│                                                                                                  │
│   186 │                                                                                          │
│   187 │   def __call__(self, *args: Any, **kwargs: Any) -> str | None:                           │
│   188 │   │   try:                                                                               │
│ ❱ 189 │   │   │   return self.function(*args, **kwargs)                                          │
│   190 │   │   except TypeError:                                                                  │
│   191 │   │   │   print(f"{self.function.__name__.strip()!r}: {self.function.__doc__.strip()}"   │
│   192 │   │   │   pwndbg.exception.handle(self.function.__name__)                                │
│                                                                                                  │
│ ------------------------/pwndbg/pwndbg/commands/dt.py:35 in dt                                   │
│                                                                                                  │
│   32 │   if address is not None:                                                                 │
│   33 │   │   address = pwndbg.commands.fix(str(address))                                         │
│   34 │                                                                                           │
│ ❱ 35 │   print(pwndbg.aglib.dt.dt(typename, addr=address))                                       │
│   36                                                                                             │
│                                                                                                  │
│ ------------------------/pwndbg/pwndbg/aglib/dt.py:75 in dt                                      │
│                                                                                                  │
│    72 │   │   header = f"{header} @ {hex(int(obj.address))}"                                     │
│    73 │   rv.append(header)                                                                      │
│    74 │                                                                                          │
│ ❱  75 │   if t.strip_typedefs().code == pwndbg.dbg_mod.TypeCode.ARRAY:                           │
│    76 │   │   return "Arrays not supported yet"                                                  │
│    77 │                                                                                          │
│    78 │   if t.strip_typedefs().code not in (                                                    │
│                                                                                                  │
│ ------------------------/pwndbg/pwndbg/dbg/gdb.py:940 in code                                    │
│                                                                                                  │
│    937 │   @property                                                                             │
│    938 │   @override                                                                             │
│    939 │   def code(self) -> pwndbg.dbg_mod.TypeCode:                                            │
│ ❱  940 │   │   assert self.inner.code in GDBType.CODE_MAPPING, "missing mapping for type code"   │
│    941 │   │   return GDBType.CODE_MAPPING[self.inner.code]                                      │
│    942 │                                                                                         │
│    943 │   @override                                                                             │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
AssertionError: missing mapping for type code
pwndbg> ptype/xo std::nullptr_t
type = void
pwndbg> 
```

### My setup
Gdb:      15.2
Python:   3.12.7
Pwndbg:   2024.08.29 build: 1c3e9507
Capstone: 5.0.1280
Unicorn:  2.1.0

--- a/pwndbg/aglib/dt.py
+++ b/pwndbg/aglib/dt.py
@@ -51,15 +51,14 @@ def dt(
         t = pwndbg.aglib.typeinfo.load(name)
 
     if not t:
-        return ""
+        return "Type not found."
 
     # If it's not a struct (e.g. int or char*), bail
-    if t.code not in (
+    if t.strip_typedefs().code not in (
         pwndbg.dbg_mod.TypeCode.STRUCT,
-        pwndbg.dbg_mod.TypeCode.TYPEDEF,
         pwndbg.dbg_mod.TypeCode.UNION,
     ):
-        return f"Not a structure: {t}"
+        return f"Not a structure: {t.strip_typedefs().name_to_human_readable}"
 
     # If an address was specified, create a Value of the
     # specified type at that address.
@@ -72,20 +71,7 @@ def dt(
         header = f"{header} @ {hex(int(obj.address))}"
     rv.append(header)
 
-    if t.strip_typedefs().code == pwndbg.dbg_mod.TypeCode.ARRAY:
-        return "Arrays not supported yet"
-
-    if t.strip_typedefs().code not in (
-        pwndbg.dbg_mod.TypeCode.STRUCT,
-        pwndbg.dbg_mod.TypeCode.UNION,
-    ):
-        newobj = obj
-        if not newobj:
-            newobj = pwndbg.dbg.selected_inferior().create_value(0, t)
-
-        iter_fields = [(field.name, field) for field in newobj.type.fields()]
-    else:
-        iter_fields = [(field.name, field) for field in t.fields()]
+    iter_fields = [(field.name, field) for field in t.fields()]
 
     for field_name, field in iter_fields:
         # Offset into the parent structure
@@ -98,17 +84,20 @@ def dt(
             pwndbg.dbg_mod.TypeCode.STRUCT,
             pwndbg.dbg_mod.TypeCode.UNION,
         ):
-            obj_value = obj[field_name]
-            if ftype.code == pwndbg.dbg_mod.TypeCode.INT:
-                extra = hex(int(obj_value))
-            elif (
-                ftype.code in (pwndbg.dbg_mod.TypeCode.POINTER, pwndbg.dbg_mod.TypeCode.ARRAY)
-                and ftype.target() == pwndbg.aglib.typeinfo.uchar
-            ):
-                data = pwndbg.aglib.memory.read(int(obj_value.address), ftype.sizeof)
-                extra = " ".join("%02x" % b for b in data)
-            else:
-                extra = obj_value.value_to_human_readable()
+            try:
+                obj_value = obj[field_name]
+                if ftype.code == pwndbg.dbg_mod.TypeCode.INT:
+                    extra = hex(int(obj_value))
+                elif (
+                    ftype.code in (pwndbg.dbg_mod.TypeCode.POINTER, pwndbg.dbg_mod.TypeCode.ARRAY)
+                    and ftype.target() == pwndbg.aglib.typeinfo.uchar
+                ):
+                    data = pwndbg.aglib.memory.read(int(obj_value.address), ftype.sizeof)
+                    extra = " ".join("%02x" % b for b in data)
+                else:
+                    extra = obj_value.value_to_human_readable()
+            except pwndbg.dbg_mod.Error as e:
+                return f"{e}\nIs the provided address near a page boundry?"
 
         # Adjust trailing lines in 'extra' to line up
         # This is necessary when there are nested structures.

--- a/pwndbg/commands/dt.py
+++ b/pwndbg/commands/dt.py
@@ -4,9 +4,7 @@ import argparse
 
 import pwndbg
 import pwndbg.aglib.dt
-import pwndbg.aglib.vmmap
-import pwndbg.color
-import pwndbg.commands
+from pwndbg.color import message
 
 parser = argparse.ArgumentParser(
     formatter_class=argparse.RawTextHelpFormatter,
@@ -16,20 +14,25 @@ parser = argparse.ArgumentParser(
     Optionally overlay that information at an address.
     """,
 )
-parser.add_argument("typename", type=str, help="The name of the structure being dumped.")
+parser.add_argument(
+    "typename",
+    type=str,
+    help='The name of the structure being dumped. Use quotes if the type contains spaces (e.g. "struct malloc_state").',
+)
 parser.add_argument(
     "address", type=int, nargs="?", default=None, help="The address of the structure."
 )
 
 
 @pwndbg.commands.ArgparsedCommand(parser)
-def dt(typename: str, address: int | pwndbg.dbg_mod.Value | None = None) -> None:
+def dt(typename: str, address: int | None = None) -> None:
     """
     Dump out information on a type (e.g. ucontext_t).
 
     Optionally overlay that information at an address.
     """
-    if address is not None:
-        address = pwndbg.commands.fix(str(address))
+    if address is not None and not pwndbg.aglib.memory.is_readable_address(address):
+        print(message.error("The provided address is not readable."))
+        return
 
     print(pwndbg.aglib.dt.dt(typename, addr=address))

